### PR TITLE
Specify a version for rapids_logger dependency

### PR DESCRIPTION
**Pull request by jlowe**
## Description
#17307 broke builds that use the rapids-cmake pinned dependencies feature since no version was specified for the rapids_logger dependency.  This adds a version string equal to the git tag so the dependency has a stated version.

## Checklist
- [X] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [X] The documentation is up to date with these changes.

### CODE REVIEW

**Review by ttnghia — APPROVED** [review]

File: cpp/CMakeLists.txt [hunk 1]
```diff
@@ -278,6 +278,7 @@ rapids_cpm_init()
 CPMAddPackage(
   NAME rapids_logger GITHUB_REPOSITORY rapidsai/rapids-logger GIT_SHALLOW TRUE GIT_TAG
   14bb233d2420f7187a690f0bb528ec0420c70d48
+  VERSION 14bb233d2420f7187a690f0bb528ec0420c70d48
```
Comment (ttnghia) [inline cpp/CMakeLists.txt:281]:
```suggestion
  14bb233d2420f7187a690f0bb528ec0420c70d48 VERSION 14bb233d2420f7187a690f0bb528ec0420c70d48
```

**Review by bdice — APPROVED** [review]
@vyasr Can you take a quick look? I am approving but I think your input would be helpful.

File: cpp/CMakeLists.txt [hunk 2]
```diff
@@ -277,7 +277,7 @@ rapids_cpm_init()
 # Not using rapids-cmake since we never want to find, always download.
 CPMAddPackage(
   NAME rapids_logger GITHUB_REPOSITORY rapidsai/rapids-logger GIT_SHALLOW TRUE GIT_TAG
-  14bb233d2420f7187a690f0bb528ec0420c70d48
+  14bb233d2420f7187a690f0bb528ec0420c70d48 VERSION 14bb233d2420f7187a690f0bb528ec0420c70d48
```
Comment (ttnghia) [inline cpp/CMakeLists.txt:280]:
```suggestion
  f9b007ae30f924de2dca8c585334f0d55331d0b8 VERSION f9b007ae30f924de2dca8c585334f0d55331d0b8
```

**Comment by ttnghia** [comment]
I've just updated the commit hash to reflect the last PR in `rapids-logger`.

**Comment by vyasr** [comment]
Wow Nghia was quick I just came here to make the same change that he did 😄

**Review by vyasr — APPROVED** [review]
Thanks! This looks fine to me for now. It's not the cleanest but we will have things in a cleaner state before the 25.02 release so I'm fine with patching in the version this way until then to ease the transition.

**Comment by vyasr** [comment]
Nghia found another minor issue with rapids-logger that I'm going to resolve now so that we can bump again in this PR.

File: cpp/CMakeLists.txt [hunk 3]
```diff
@@ -277,7 +277,7 @@ rapids_cpm_init()
 # Not using rapids-cmake since we never want to find, always download.
 CPMAddPackage(
   NAME rapids_logger GITHUB_REPOSITORY rapidsai/rapids-logger GIT_SHALLOW TRUE GIT_TAG
-  14bb233d2420f7187a690f0bb528ec0420c70d48
+  f9b007ae30f924de2dca8c585334f0d55331d0b8 VERSION f9b007ae30f924de2dca8c585334f0d55331d0b8
```
Comment (vyasr) [inline cpp/CMakeLists.txt:280]: Temporarily changing this to verify that things work as expected.
```suggestion
  fix/include_dir VERSION f9b007ae30f924de2dca8c585334f0d55331d0b8
```

File: cpp/CMakeLists.txt [hunk 4]
```diff
@@ -277,7 +277,7 @@ rapids_cpm_init()
 # Not using rapids-cmake since we never want to find, always download.
 CPMAddPackage(
   NAME rapids_logger GITHUB_REPOSITORY rapidsai/rapids-logger GIT_SHALLOW TRUE GIT_TAG
```
Comment (vyasr) [inline cpp/CMakeLists.txt:279]:
```suggestion
  NAME rapids_logger GITHUB_REPOSITORY vyasr/rapids-logger GIT_SHALLOW TRUE GIT_TAG
```

**Comment by vyasr** [comment]
@ttnghia I pushed a fix to https://github.com/rapidsai/rapids-logger/pull/3/. Could you rebase your local changes onto Jason's cudf branch's latest commit and make sure that things build as you expect (you can link to `cudf::cudf_logger` without linking to `cudf::cudf`)?

File: cpp/CMakeLists.txt [hunk 5]
```diff
@@ -276,8 +276,8 @@ rapids_cpm_init()
 
 # Not using rapids-cmake since we never want to find, always download.
 CPMAddPackage(
-  NAME rapids_logger GITHUB_REPOSITORY rapidsai/rapids-logger GIT_SHALLOW TRUE GIT_TAG
-  14bb233d2420f7187a690f0bb528ec0420c70d48
+  NAME rapids_logger GITHUB_REPOSITORY vyasr/rapids-logger GIT_SHALLOW TRUE GIT_TAG
+  fix/include_dir VERSION f9b007ae30f924de2dca8c585334f0d55331d0b8
```
Comment (vyasr) [inline cpp/CMakeLists.txt:280]:
```suggestion
  NAME rapids_logger GITHUB_REPOSITORY vyasr/rapids-logger GIT_SHALLOW TRUE GIT_TAG fix/include_dir
  VERSION f9b007ae30f924de2dca8c585334f0d55331d0b8
```

**Comment by vyasr** [comment]
CI is passing here. Once @ttnghia can confirm that the latest version fixes his issue we can merge the rapids-logger PR, update the commit hash in this PR (and revert the repo changes) then rerun and merge.

**Comment by ttnghia** [comment]
I confirm that the patch fixes all the previous bugs. `spark-rapids-jni` now can build without issue.

**Comment by bdice** [comment]
This fix appears to be necessary for devcontainers, too. I am getting
```
[0/8] Performing download step (git clone) for 'rapids_logger-populate'
Cloning into 'rapids_logger-src'...
fatal: unable to read tree (14bb233d2420f7187a690f0bb528ec0420c70d48)
CMake Error at rapids_logger-subbuild/rapids_logger-populate-prefix/tmp/rapids_logger-populate-gitclone.cmake:61 (message):
  Failed to checkout tag: '14bb233d2420f7187a690f0bb528ec0420c70d48'
```

but it builds successfully with this PR.

**Comment by davidwendt** [comment]
What happens here next? Is this dependent on merging https://github.com/rapidsai/rapids-logger/pull/3 ?

**Comment by bdice** [comment]
> What happens here next? Is this dependent on merging [rapidsai/rapids-logger#3](https://github.com/rapidsai/rapids-logger/pull/3) ?

Yes, I merged that fix and updated the commit hash here. If we get through a few CI builds, I think this will be good enough to admin-merge.

**Pull request merged**